SimCalorimetry/EcalEBTrigPrimAlgos/src/EcalEBTrigPrimClusterAlgo.cc

```diff
@@ -0,0 +1,469 @@
+	    }
+	    std::cout<<std::endl;
+	  }
+
+	  // call peakfinder
+	  this->getPeakFinder()->process(filt_out_,peak_out_);
+ 
+	  if(debug_){
+	    std::cout<< "output of peakfinder is a vector of size: "<<std::dec<<peak_out_.size()<<std::endl; 
+	    for (unsigned int ix=0;ix<peak_out_.size();ix++){
+	      std::cout<<std::dec<<peak_out_[ix] << " " ;
+	    }
+	    std::cout<<std::endl;
+	  }
+
+	  // call formatter
+	  this->getFormatterEB()->setParameters(stripid,ecaltpgSlidW_) ; 
+	  this->getFormatterEB()->process(fgvb_out_,peak_out_,filt_out_,format_out_); 
+
+	  if (debug_) {
+	    std::cout<< "output of formatter is a vector of size: "<<format_out_.size()<<std::endl; 
+	    for (unsigned int i =0; i<format_out_.size();i++){
+	      std::cout <<" "<<std::dec<<format_out_[i] << " " ;
+	    }    
+	    std::cout<<std::endl;
+	  }
+	  
+
+	  // call final tcp formatter
+	  this->getFormatter()->setParameters( thisTower.rawId(),ecaltpgLutGroup_,ecaltpgLut_,ecaltpgBadTT_,ecaltpgSpike_);
+	  this->getFormatter()->process(format_out_,tcpformat_out_);
+
+	  std::vector<SimpleCaloHit> singleChan;
+	  for (int iSample=firstSample_;iSample<=lastSample_;++iSample) {	
+            SimpleCaloHit singleSample(0);
+	    etInADC= tcpformat_out_[iSample];
+	    singleSample.setEtInADC(etInADC);
+            singleSample.setId(myid);
+            singleSample.setPosition(GlobalVector( theBarrelGeometry->getGeometry(myid)->getPosition().x(), 
+						   theBarrelGeometry->getGeometry(myid)->getPosition().y(), 
+						   theBarrelGeometry->getGeometry(myid)->getPosition().z()));
+	    //	    std::cout << " iSample " << iSample << " et ADC " << etInADC << " ID " << myid << " position " << singleSample.position() << std::endl;	    
+
+	    singleChan.push_back(singleSample);
+	  }
+	  hitCollection.push_back(singleChan); 
+	  
+
+	} // Loop over the xStals in  a strip
+
+      }//loop over strips in one tower
+
+  }   // loop over the towers 
+
+  
+  std::vector<uint16_t> cluCollection = makeCluster ( hitCollection, result, dEta, dPhi, hitNoiseCut, etCutOnSeed  );
+  if (debug_) std::cout << "  TrigPrimClusterAlgo hitCollection size " << hitCollection.size() << " cluster size " << cluCollection.size() <<  std::endl; 
+
+  /*
+  for (unsigned int iclu=0; iclu < cluCollection.size(); iclu++) {
+    int nSam=0;
+    bool isASpike=0; 
+    int timing=0;
+    
+    for (int iSample=firstSample_;iSample<=lastSample_;++iSample) {
+     
+      if ( iSample!=5 ) {
+        etInADC = 0;
+      } else {
+	etInADC = cluCollection[iclu];
+      }
+      tp.setSample(nSam,  EcalEBClusterTriggerPrimitiveSample(etInADC,isASpike,timing)  ); 
+     nSam++;
+    }
+
+    if (!tcpFormat_)
+      result.push_back(tp);
+    else 
+      resultTcp.push_back(tp);
+
+  } // loope over the clusters
+  */
+
+}
+
+
+
+std::vector<uint16_t>  EcalEBTrigPrimClusterAlgo::makeCluster ( std::vector<std::vector<SimpleCaloHit> >&  hitCollection, 
```

Comment:
I think `hitCollection` could be passed as const ref

Reply:
So?